File: src/main/java/team4/footwithme/vote/repository/CustomChoiceRepository.java
package team4.footwithme.vote.repository;

import team4.footwithme.vote.domain.Choice;

import java.util.List;

public interface CustomChoiceRepository {

    Long countByVoteItemId(Long voteItemId);

    List<Choice> findByMemberIdAndVoteId(Long memberId, Long voteId);

    List<Long> findMemberIdsByVoteItemId(Long voteItemId);

    Long maxChoiceCountByVoteId(Long voteId);

}

File: src/main/java/team4/footwithme/vote/repository/CustomChoiceRepository.kt
package team4.footwithme.vote.repository

import team4.footwithme.vote.domain.Choice

interface CustomChoiceRepository {
    fun countByVoteItemId(voteItemId: Long?): Long?

    fun findByMemberIdAndVoteId(memberId: Long?, voteId: Long?): List<Choice>?

    fun findMemberIdsByVoteItemId(voteItemId: Long?): List<Long>?

    fun maxChoiceCountByVoteId(voteId: Long?): Long
}
